### ISSUE TITLE
Add search in facet method on the helper.

### ISSUE THREAD

**Issue by bobylito** [algolia/algoliasearch-helper-js#420]
# API proposal:

```js
helper = algoliasearchHelper('appId', 'apiKey', 'indexName' {
  facets: ['a']
});
helper.searchInFacets('a', 'blah').then(resultsCallback); 
```

this does a search in facet for the current index using the attribute `a` and the search query `blah`

## results callback options?

 - Unify the structure with the one found on the facets currently in the helper?
 - only provide the array?

**Comment by vvo** [comment]
Nice!

I have trouble understanding the `results callback options?` maybe detail a bit more what you have in mind?

Also it should be searchForFacetValues (and yes we have to update it in the JS client too, we can rename it actually, it's not officially out).

**Comment by bobylito** [comment]
Ah I wasn't aware of the change of name, it is fine with me :)

> I have trouble understanding the results callback options? maybe detail a bit more what you have in mind?

The correct question is: What are the parameters that the `resultsCallback` function (in the example) will receive when called? Proposal:

```js
function resultsCallback(results) {
  console.log(results)
  // output:
  // [ {value: 'a', highlighted: 'a', count: 10}, {value: 'b', highlighted: 'b', count: 5} ]
}
```

**Comment by bobylito** [comment]
It also needs to send the parameters from the state to send the correct context to the search for facets values.

**Comment by vvo** [comment]
> It also needs to send the parameters from the state to send the correct context to the search for facets values.

Yes this is the true value of adding this inside the helper directly.

As for the function API I am more and more keen to always provide Object style API:

```js
searchForFacetValues({
  query: '...',
  facetName: '...'
});
```

What do you think?

**Comment by bobylito** [comment]
> As for the function API I am more and more keen to always provide Object style API:

On one hand I would say that since it's close to `searchOnce` this kind of API make sense but on the other hand those two parameters are mandatory and the helper API is more parameter based, so I would stick to parameters not to objects.

Also that makes me think, that we might want to override the parameters sent to Algolia. What about adding a third parameter that can override the values?

```js
helper.searchForFacetValues('facet', 'query', {
  nbHits: 3
});
```

I don't know if the number of hits is compatible :)

**Comment by vvo** [comment]
I do not yet know if there's a need to be able to override the current state of the helper as for search for facet values.

**Comment by vvo** [comment]
I would go maybe for:

```js
helper.searchForFacetValues('facetName', {
  query: ..
});
```

That way you can just merge current search params with the second argument :heavy_check_mark:.

It's also in line with `getFacetValues('facetName', options)`, what do you think?

**Comment by bobylito** [comment]
Released in 2.16.0
